### ISSUE TITLE
Add a way to "recover" from the "wrong" repository

### ISSUE THREAD

**Issue by colearendt** [rstudio/renv#504]
I have had it happen several times where I:

- am restoring an environment and want to be sure to use binary packages from RSPM
- start `renv::restore()`
- see packages downloading from `cran.rstudio.com` or some other repo with no binaries
- CTRL+C to abort
- fix the repository URL
- restart `renv::restore()`
- `renv` _still_ installs the downloaded packages from the first pass (i.e. source bundles) and does not go back to retrieve them

It would be great if there were a way to gracefully "abort" and "re-restore" from a fresh state.

**Comment by kevinushey** [comment]
One solution: we could stop keeping a local copy of source packages. (If we're already caching and sharing installed packages, it seems unnecessary to keep binaries + sources locally as well anyhow)

We could make this configurable by an `renv` option (e.g. `renv.config.cache.sources = FALSE` or something).

**Comment by donarus** [comment]
Voting for this ..

Our use case:
* we use version 99.99.99.9000 as a snapshot/dev version and deploy these snapshot artifacts to our internal CRAN repository.
* time to time we (developers) need to reinstall these snapshot packages on our computers to get the latest unreleased features for our other packages (also under development)
* this was easily possible with packrat just by calling install.packages('xxx') and it worked. But we are not able to do the same with RENV. What we have to do is to manually clean the "repository" folder, which is annoying


Proposed solution

A] (https://github.com/rstudio/renv/blob/25ed22481bdeb89b5dde5f9ab5bc971597b14c4c/R/download.R#L46-L53) there is a check that file is an archive in the **_download_** method and if so, then nothing else is checked. The simplest solution would be to remove this check and just compare at least the size of the file with the size reported in HTTP header. 

B] add option **_keepSources_** to "purge" method with defaults to FALSE and handle the deletion directly in the PURGE method.

I tried to prepare the PR but due to lack of time I was not able to get through all the "renv_retrieve_xxxx" and "renv_install_xxxx" methods

**Comment by donarus** [comment]
And yes, this seems to be a real blocker for projects with more complex internal dependencies.

**Comment by kevinushey** [comment]
I agree with your first proposal; that feels like a misguided optimization.

**Comment by hadley** [comment]
Looks like the download size is checked, https://github.com/rstudio/renv/blob/b4493f95b0d73a370ce3a5cac39cb07a73e7ee12/R/download.R#L54-L64, so this can be closed.
